### PR TITLE
initial commit of new chat feature (fixes #2271)

### LINKED ISSUE #2257: chat: submitting a blank field or newline crashed app
submitting a chat prompt while the field is empty or with some text and a newline leads to the app crashing

### LINKED ISSUE #2271: create a chat experience to our chatapi

### PR DESCRIPTION
**Pull request by Okuro3499**
In this pull request, we introduce an exciting enhancement that integrates a new chat API endpoint. The new functionality allows users to engage in real-time chat conversations with Chat Api within myPlanet

fixes #2257

### CODE REVIEW

File: app/src/main/res/values-fr/strings.xml [hunk 1]
```diff
@@ -896,6 +896,17 @@
...
+    <string name="request_to_join">Request To join</string>
+    <string name="filter_labels">Filter Labels</string>
+    <string name="message">message</string>
+    <string name="mistakes">Mistakes</string>
+    <string name="take_survey">Take Survey</string>
+    <string name="checkbox">CheckBox</string>
+    <string name="offer">Offer</string>
+    <string name="request_for_advice">Request for advice</string>
+    <string name="ai_chat">AI Chat</string>
+>>>>>>> chatFeature
```
Comment (rrijal53) [inline app/src/main/res/values-fr/strings.xml:909]: Need to fix this, seems like you had an issue when rebasing the branch.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -985,6 +985,9 @@
         It has been used to improve early education, secondary schools, village health, youth workforce development,
         and economic and community development.</p>
     <p>Planet houses is a repository of free, open access and public domain resources to benefit all learners.</p>
+=======
+    <string name="ai_chat">AI Chat</string>
+>>>>>>> chatFeature
```
Comment (rrijal53) [inline app/src/main/res/values/strings.xml:990]: same as above

File: app/src/main/res/values-es/strings.xml [hunk 3]
```diff
@@ -896,6 +896,9 @@
         Se ha utilizado para mejorar la educación temprana, las escuelas secundarias, la salud en las aldeas, el desarrollo de la fuerza laboral juvenil,
         y el desarrollo económico y comunitario.</p>
     <p>Planet alberga un repositorio de recursos gratuitos, de acceso abierto y de dominio público para beneficiar a todos los estudiantes.</p>
+=======
+    <string name="ai_chat">AI Chat</string>
+>>>>>>> chatFeature
```
Comment (rrijal53) [inline app/src/main/res/values-es/strings.xml:901]: fix this merge conflict.

File: app/src/main/res/layout/item_ai_response_message.xml [hunk 4]
```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_marginEnd="35dp"
+    android:layout_marginStart="8dp"
+    android:layout_marginTop="16dp">
+
+    <!--    <TextView-->
```
Comment (rrijal53) [inline app/src/main/res/layout/item_ai_response_message.xml:10]: remove comment if not required

File: app/src/main/java/org/ole/planet/myplanet/utilities/Utilities.java [hunk 5]
```diff
@@ -126,6 +126,22 @@ public static String getUrl() {
...
 
+    public static String getHostUrl() {
+        SharedPreferences settings = MainApplication.context.getSharedPreferences(PREFS_NAME, MODE_PRIVATE);
+        if (settings.contains("url_Host")) {
+            String scheme = settings.getString("url_Scheme", "");
+            String hostIp = settings.getString("url_Host", "");
+            String url = scheme +"://"+ hostIp;
+            Log.d("url_Host", url);
+
+//            if (!url.endsWith("/db")) {
```
Comment (rrijal53) [inline app/src/main/java/org/ole/planet/myplanet/utilities/Utilities.java:137]: Is this required?

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (rrijal53) [inline app/src/main/res/values/strings.xml:990]: remove this

**Review by Mutugiii — APPROVED** [review]
The functionality is great! I tested it thoroughly and was unable to find any issues.

In terms of the layout, the virtual keyboard on devices blocks the form field

**Review by dogi — APPROVED** [review]

**Pull request merged**